src/accessory/HumidifierAccessory.ts

```diff
@@ -0,0 +1,105 @@
+import { PlatformAccessory } from 'homebridge';
+import { TuyaPlatform } from '../platform';
+import BaseAccessory from './BaseAccessory';
+
+export default class HumidifierAccessory extends BaseAccessory {
+
+  constructor(platform: TuyaPlatform, accessory: PlatformAccessory) {
+    super(platform, accessory);
+
+    this.configureActive();
+    this.configureTargetState();
+    this.configureCurrentState();
+    this.configureCurrentRelativeHumidity();
+    this.configureRelativeHumidityHumidifierThreshold();
+    this.configureTemperatureSensor();
+  }
+
+  mainService() {
+    return this.accessory.getService(this.Service.HumidifierDehumidifier)
+      || this.accessory.addService(this.Service.HumidifierDehumidifier);
+  }
+
+  configureActive(){
+    const { ACTIVE, INACTIVE } = this.Characteristic.Active;
+    this.mainService().getCharacteristic(this.Characteristic.Active)
+      .onGet(() => {
+        const status = this.getStatus('switch');
+        return (status?.value as boolean) ? ACTIVE : INACTIVE;
+      })
+      .onSet(value => {
+        this.sendCommands([{ code: 'switch', value: (value === ACTIVE) ? true : false }]);
+      });
+  }
+
+  configureTargetState(){
+    const { HUMIDIFIER} = this.Characteristic.TargetHumidifierDehumidifierState;
+    const validValues = [ HUMIDIFIER ];
+
+    this.mainService().getCharacteristic(this.Characteristic.TargetHumidifierDehumidifierState)
+      .onGet(() => {
+        return HUMIDIFIER;
+      }).setProps({ validValues });
+  }
+
+  configureCurrentState(){
+    const { INACTIVE, HUMIDIFYING } = this.Characteristic.CurrentHumidifierDehumidifierState;
+
+    this.mainService().getCharacteristic(this.Characteristic.CurrentHumidifierDehumidifierState)
+      .onGet(() => {
+        const status = this.getStatus('switch');
+        return (status?.value as boolean) ? HUMIDIFYING : INACTIVE;
+      });
+  }
+
+  configureCurrentRelativeHumidity(){
+    this.mainService().getCharacteristic(this.Characteristic.CurrentRelativeHumidity)
+      .onGet(() => {
+        const status = this.getStatus('humidity_current');
+        const humidity = Math.min(100, Math.max(0, status?.value as number));
+        return humidity;
+      });
+  }
+
+  configureRelativeHumidityHumidifierThreshold(){
+    const schema = this.getSchema('humidity_set');
+    const s = JSON.parse(schema?.values as string);
```

Comment:
Please use `schema.property` instead,  it already parsed from json. `schema.values` is kept for compatible of the legacy accessories, and it will be removed in the future.

Reply:
Thanks, fixed